server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java

```diff
@@ -77,11 +77,6 @@ public void evaluate(ResourcePermission permission, EvaluationContext executionC
             if (resource.getType() != null) {
                 evaluatePolicies(() -> policyStore.findByResourceType(resource.getType(), resourceServer.getId()), consumer);
             }
-
-            if (scopes.isEmpty() && !resource.getScopes().isEmpty()) {
-                scopes.removeAll(resource.getScopes());
-                evaluatePolicies(() -> policyStore.findByScopeIds(resource.getScopes().stream().map(Scope::getId).collect(Collectors.toList()), resourceServer.getId()), consumer);
-            }
         }
 
         if (!scopes.isEmpty()) {
```

Comment:
@pedroigor 

We identified that this block is also responsible for performances issues when doing entitlement or evaluation when many resources share the same scopes